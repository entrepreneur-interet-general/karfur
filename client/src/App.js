import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Loadable from 'react-loadable';
import { Provider } from 'react-redux';
import track from 'react-tracking'; 
import { Spinner } from 'reactstrap';
import IdleTimer from 'react-idle-timer'
import './scss/fonts/circular-std/css/circular-std.css'

import Store from './Store/configureStore';
import { PrivateRoute } from './components/PrivateRoute';
import { socket } from './utils/API';
import {dispatch} from './tracking/dispatch';
import './i18n';

import 'react-notifications/src/notifications.scss';
import './App.scss';

const loading = () => <Spinner color="success" />;
const chargement = () => <div>Chargement</div>

// Containers
const Layout = Loadable({
  loader: () => import('./containers/Layout/Layout'),
  loading
});

// Pages
const Login = Loadable({
  loader: () => import('./containers/Login/Login'),
  loading
});


const Register = Loadable({
  loader: () => import('./containers/Register/Register'),
  loading
});

const Page404 = Loadable({
  loader: () => import('./views/Pages/Page404'),
  loading
});

const Page500 = Loadable({
  loader: () => import('./views/Pages/Page500'),
  loading
});

const LiveChat = Loadable({
  loader: () => import('./components/UI/LiveChat/LiveChat'),
  loading : chargement
});

class App extends Component {
  state = { data: {} }
  idleTimer = null;

  componentDidMount() {    
    socket.on('server:event', data => {
      console.log('Ã©vÃ¨nement',data)
      this.setState({ data })
    })
    window.onbeforeunload = function() {
      this.props.tracking.trackEvent({ action: 'unmount', label: 'App' });
      return undefined;
    }.bind(this);
  }

  // _onAction = (e) => {
  //   // console.log('user did something', e)
  //   return;
  // }

  _onActive = () => this.props.tracking.trackEvent({ action: 'active', label: 'App', value: this.idleTimer.getRemainingTime() });

  _onIdle = () => this.props.tracking.trackEvent({ action: 'idle', label: 'App', value: this.idleTimer.getLastActiveTime() });

  sendMessage = (message,side) => {
    socket.emit(side + ':sendMessage', message)
  }
  socketFn={
    sendMessage: this.sendMessage
  }

  render() {
    return (
      <Provider store={Store}>
        <IdleTimer
          ref={ref => { this.idleTimer = ref }}
          element={document}
		      onActive={this._onActive}
          onIdle={this._onIdle}
          // onAction={this._onAction}
		      // debounce={250}
          timeout={1000 * 60 * 5} />
        <BrowserRouter>
            <Switch>
              <Route exact path="/login" name="Login Page" component={Login} />
              <Route exact path="/register" name="Register Page" component={Register} />
              <Route exact path="/404" name="Page 404" component={Page404} />
              <Route exact path="/500" name="Page 500" component={Page500} />
              <PrivateRoute 
                  path='/' 
                  component={Layout} 
                  socket = { socket } 
                  socketFn = { this.socketFn }/>
            </Switch>
        </BrowserRouter>
        <LiveChat socket = { socket } 
                  socketFn = { this.socketFn } /> 
      </Provider>
    );
  }
}

export default track({
  app: 'App',
}, { dispatch: dispatch, dispatchOnMount: true, process: (ownTrackingData) => ownTrackingData.page ? true : null })(
  App
)
