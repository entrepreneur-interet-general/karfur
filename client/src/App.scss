// Styles
// CoreUI Icons Set
@import '~@coreui/icons/css/coreui-icons.css';
// Import Flag Icons Set
@import '~flag-icon-css/css/flag-icon.min.css';
// Import Font Awesome Icons Set
@import '~font-awesome/css/font-awesome.min.css';
// Import Simple Line Icons Set
@import '~simple-line-icons/css/simple-line-icons.css';
// Import Main styles for this application
@import './scss/style.scss';

body,head{
  font-family: CircularStd;
  color: #3D3D3D;
  height:100%;
  min-height:100vh;

  [contenteditable] {
    -webkit-user-select: text;
    user-select: text;
  }
}

.spinner-container{
  display:flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height:100vh;
  width:100vw;
}