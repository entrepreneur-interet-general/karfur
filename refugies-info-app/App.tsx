import React from "react";
import MainApp from "./src/App";
// import i18n (needs to be bundled ;))
import "./i18n";

export default function App() {
  return <MainApp />;
}
