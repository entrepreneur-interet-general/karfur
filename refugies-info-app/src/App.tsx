import { StatusBar } from "expo-status-bar";
import React, { useState, useEffect } from "react";
import { SafeAreaProvider } from "react-native-safe-area-context";

import useCachedResources from "./hooks/useCachedResources";
import useColorScheme from "./hooks/useColorScheme";
import Navigation from "./navigation";
import { AsyncStorage } from "react-native";
import i18n from "./services/i18n/index";

export default function App() {
  const isLoadingComplete = useCachedResources();
  const colorScheme = useColorScheme();
  const [isI18nInitialized, setIsI18nInitialized] = useState(false);

  useEffect(() => {
    const setLanguage = async () => {
      try {
        await i18n.init();

        try {
          const value = await AsyncStorage.getItem("SELECTED_LANGUAGE");
          if (value) {
            i18n.changeLanguage(value);

            // value previously stored
          }
        } catch (e) {
          // error reading value
        }
        setIsI18nInitialized(true);
      } catch (error) {
        console.warn(error);
      }
    };
    setLanguage();
  }, []);

  if (!isLoadingComplete || !isI18nInitialized) {
    return null;
  }

  return (
    <SafeAreaProvider>
      <Navigation colorScheme={colorScheme} />
      <StatusBar />
    </SafeAreaProvider>
  );
}
