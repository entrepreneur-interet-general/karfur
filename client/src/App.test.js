import { expect } from 'chai'
import React from 'react'
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'

import App from './App'

describe('App', () => {
  it('mounts without crashing', () => {
    const jsdomScroll = window.scrollTo;  // remember the jsdom alert
    window.scrollTo = () => {};  // provide an empty implementation for window.alert
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
    window.scrollTo = jsdomScroll;  // restore the jsdom alert
  });

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<App debug />);
  });

  it('should match its reference snapshot', () => {
    const wrapper = shallow(<App />)
  
    expect(wrapper).to.matchSnapshot()
  })

  // it('renders welcome message', function() {
  //   const wrapper = shallow(<App />); 
  //   const welcome = <h1>Bienvenue dans le projet Karfu'R</h1>;
  //   expect(wrapper.hasClass('hero-container')).to.equal(true);
  // });

  // it('calls componentDidMount', () => {
  //   jest.spyOn(App.prototype, 'componentDidMount')
  //   const wrapper = shallow(<App />)
  //   expect(App.prototype.componentDidMount.mock.calls.length).to.be(1)
  // })
})