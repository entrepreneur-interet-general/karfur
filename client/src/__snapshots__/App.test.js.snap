// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match its reference snapshot 1`] = `
<App
  tracking={
    Object {
      "getTrackingData": [Function],
      "trackEvent": [Function],
    }
  }
/>
`;
